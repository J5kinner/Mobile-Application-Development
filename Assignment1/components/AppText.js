import React from 'react';

import { Text, StyleSheet, Platform } from 'react-native';

function AppText({style, children}) {
    return (
        <Text style={[styles.container, style]}> {children} </Text>
    );
}

const styles = StyleSheet.create({
    text:{
        fontSize:100,
        fontFamily: Platform.OS === 'android' ? "monospace" : "Cochin",

    }
})

export default AppText;