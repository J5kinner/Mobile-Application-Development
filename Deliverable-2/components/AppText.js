import React from "react";

import { Text, StyleSheet, Platform } from "react-native";

function AppText({ style, children }) {
  return <Text style={[styles.text, style]}> {children} </Text>;
}

const styles = StyleSheet.create({
  text: {
    fontSize: 21,
    fontFamily: Platform.OS === "android" ? "System" : "Avenir-Roman",
  },
});

export default AppText;
