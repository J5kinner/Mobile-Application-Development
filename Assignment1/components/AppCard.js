import React from 'react';
import { View, Image, Text, StyleSheet } from 'react-native';
import AppColors from '../config/AppColors';
import AppText from './AppText';


function AppCard({title, subtitle, image}) {
    return (
        <View style={styles.container}>
            <Image style={styles.image} source={image}/>
            <AppText style={styles.title}>{title}</AppText>
            <AppText style={styles.subtitle}>{subtitle}</AppText>

            
        </View>
    );

}

const styles = StyleSheet.create({
    container:{
        backgroundColor: AppColors.inputTextBackground,
        borderRadius: 25,
        overflow:"hidden",
        marginBottom: 25,
    },
    image:{
        height: 200, 
        width:"100%",
    },
    title:{
        fontWeight: "bold",

    },
    subtitle:{
        fontSize: 16,
    },

    
})

export default AppCard;