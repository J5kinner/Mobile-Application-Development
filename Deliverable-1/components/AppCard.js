import React from "react";
import { View, Image, StyleSheet } from "react-native";
import AppColors from "../config/AppColors";
import AppText from "./AppText";

function AppCard({ category, title, subtitle, image }) {
  return (
    <View style={styles.container}>
      {isFinite(image) ? (
        <Image source={image} style={styles.image} />
      ) : (
        <Image style={styles.image} source={{ uri: image }} />
      )}
      <AppText style={styles.title}>{title}</AppText>
      <AppText style={styles.subtitle}>{subtitle}</AppText>
      <AppText style={styles.subtitle}>{category}</AppText>
      
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: AppColors.inputTextBackground,
    borderRadius: 25,
    marginBottom: 25,
    overflow: "hidden",
  },
  image: {
    height: 200,
    width: "100%",
  },
  title: {
    fontWeight: "bold",
  },
  subtitle: {
    fontSize: 16,
    padding: 5,
  },
});

export default AppCard;
