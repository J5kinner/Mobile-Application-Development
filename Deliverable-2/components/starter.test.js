import React from "react";
import renderer from "react-test-renderer";

import AppText from "../components/AppText.js";
import AppCard from "./AppCard.js";

jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");

describe("Testing react rendering of components", () => {
  test("renders correctly", async () => {
    const json = renderer.create(<AppText />).toJSON();
    expect(json).toMatchSnapshot();
  });
  test("App Cards are rendered correctly with corresponding content", async () => {
    const component = renderer
      .create(
        <AppCard
          category="adventure"
          title="Adventure Park"
          subtitle="Forest theme park for exploring the wilderness"
          image="./assets/user4.jpg"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

test("AppText rendering", () => {
  const json = renderer.create(<AppText />).toJSON();
  expect(json.props.style[0].fontSize).toBe(21);
  expect(json.props.style[0].fontFamily).toBe("Avenir-Roman");
});

test("AppText contains text", () => {
  const json = renderer.create(<AppText>Hello World</AppText>).toJSON();
  expect(json.props.style[0].fontSize).toBe(21);
  expect(json.props.style[0].fontFamily).toBe("Avenir-Roman");
  expect(json.children.includes("Hello World"));
});
