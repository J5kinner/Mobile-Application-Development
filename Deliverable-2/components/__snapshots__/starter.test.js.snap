// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react rendering of components App Cards are rendered correctly with corresponding content 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e3e2e1",
      "borderRadius": 25,
      "marginBottom": 25,
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "./assets/user4.jpg",
      }
    }
    style={
      Object {
        "height": 200,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 21,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
     
    Adventure Park
     
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 21,
        },
        Object {
          "fontSize": 16,
          "padding": 5,
        },
      ]
    }
  >
     
    Forest theme park for exploring the wilderness
     
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 21,
        },
        Object {
          "fontSize": 16,
          "padding": 5,
        },
      ]
    }
  >
     
    adventure
     
  </Text>
</View>
`;

exports[`Testing react rendering of components AppText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Avenir-Roman",
        "fontSize": 21,
      },
      undefined,
    ]
  }
>
   
   
</Text>
`;
